src/lib/app/mainapplication.cpp

```diff
@@ -745,13 +746,17 @@ void MainApplication::saveSettings()
     if (deleteCookies) {
         m_cookieJar->deleteAllCookies();
     }
+    if (deleteCache) {
+        DataPaths::clearCacheData();
```

Comment:
A change here may affect OS X... did notice it killed the entire `.../qupzilla/cache` folder which I thought was a little odd but it did work. I'll keep my eye out for it.